include/libtorrent/alert_types.hpp

```diff
@@ -56,6 +56,9 @@ POSSIBILITY OF SUCH DAMAGE.
 #pragma GCC diagnostic ignored "-Wdeprecated-declarations"
 #endif
 
+// c++/cx keyword
+#undef interface
```

Comment:
I remember I asked you this in your first version of this patch too. if "interface" really is a keyword, #undef won't do anything. If it is an actual define, won't this interfere with other c++/cx code that may come after including this file?
Isn't there a cleaner way to mix c++ and c++/cx? something like extern "C++" { ... } around the whole header?